core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationStatusService.js

```diff
@@ -19,35 +19,99 @@
+    var explorationContentRequiredCount = 0;
+    var explorationContentNotAvailableCount = 0;
     var contentIdsToAudioTranslations =
         StateContentIdsToAudioTranslationsService.displayed;
-    var _computeAllStatesStatus = function() {
+    var writtenTranslations = StateWrittenTranslationsService.dsiplayed;
+
+    var _getVoiceOverStatus = function(
+        contentIdsToAudioTranslations, contentId) {
+      // The first value of availabilityStatus shows whether a audio is
+      // available for a contentId in a given language, whereas the last value
+      // represent whether the available audio needs update.
+      var availabilityStatus = [false, false];
+      var availableLanguage = (
+        contentIdsToAudioTranslations.getAudioLanguageCodes(contentId));
+      if (availableLanguage.indexOf(langCode) !== -1) {
+        availabilityStatus[0] = true;
+        var audioTranslation = (
+          contentIdsToAudioTranslations.getAudioTranslation(
+            contentId, langCode));
+        availabilityStatus[1] = audioTranslation.needsUpdate;
+      }
+      return availabilityStatus;
+    };
+
+    var _getTranslationStatus = function(writtenTranslations, contentId) {
+      // The first value of availabilityStatus shows whether a translation is
+      // available for a contentId in a given language, whereas the last value
+      // represent whether the available translation needs update.
+      var availabilityStatus = [false, false];
+      var availableLanguage = (
+        writtenTranslations.getTranslationsLanguageCodes(contentId));
+      if (availableLanguage.indexOf(langCode) !== -1) {
+        var writtenTranslation = (
+          writtenTranslations.getWrittenTranslation(contentId, langCode));
+        if (writtenTranslation.getHtml() !== '') {
+          availabilityStatus[0] = true;
+          availabilityStatus[1] = writtenTranslation.needsUpdate;
+        }
+      }
+      return availabilityStatus;
+    };
+    var _getContentAvailabilityStatus = function(stateName, contentId) {
       langCode = TranslationLanguageService.getActiveLanguageCode();
+      if (TranslationTabActiveModeService.isTranslationModeActive()) {
+        var writtenTranslations = (
+          ExplorationStatesService.getWrittenTranslationsMemento(stateName));
+        return _getTranslationStatus(writtenTranslations, contentId);
+      } else if (TranslationTabActiveModeService.isVoiceoverModeActive()) {
+        var contentIdsToAudioTranslations = (
+          ExplorationStatesService.getContentIdsToAudioTranslationsMemento(
+            stateName));
+        return _getVoiceOverStatus(contentIdsToAudioTranslations, contentId);
+      }
+    };
+
+    var _getActiveStateContentAvailabilityStatus = function(contentId) {
+      if (TranslationTabActiveModeService.isTranslationModeActive()) {
+        var writtenTranslations = StateWrittenTranslationsService.displayed;
+        return _getTranslationStatus(writtenTranslations, contentId);
```

Comment:
what if writtenTranslations is empty?

Reply:
Basically, can a situation arise when after line 93, the writtenTranslations object is None/undefined? If so, would that lead to breakages when this function is called?

Reply:
TranslationStatusService expects that StateWrittenTranslationsService.displayed is always defined and if it's not defined then there is some issue with the system.